src/scripts/towns/ProfOakNPC.ts

```diff
@@ -0,0 +1,41 @@
+class ProfOakNPC extends NPC {
+
+    constructor(
+        public name: string,
+        public dialog: string[]
+    ) {
+        super(name, dialog);
+    }
+
+    get dialogHTML(): string {
+        const UniquePokemonCaught = App.game.party.caughtPokemon.filter(p => p.id > 0).length;
```

Comment:
This won't quite work as expected, as different forms would also attribute to the `UniquePokemonCaught` count.
```suggestion
        const UniquePokemonCaught = new Set(App.game.party.caughtPokemon.filter(p => p.id > 0).map(p => Math.floor(p.id))).size;
```

Can we also change `UniquePokemonCaught` → `uniquePokemonCaught` as we are mostly only using PascalCase for Global variables.